packages/animated-qr/src/AnimatedQRScanner/BaseQRScanner.tsx

```diff
@@ -32,36 +36,48 @@ export const BaseQRScanner = ({
   }, []);
 
   useEffect(() => {
-    const videoElement = document.getElementById(VIDEO_ID) as HTMLVideoElement;
+    const videoElement = document.getElementById(
+      VIDEO_ID
+    ) as HTMLVideoElementScanPreview;
     const canplayListener = () => {
       setCanplay(true);
       videoLoaded && videoLoaded(true);
     };
     videoElement.addEventListener("canplay", canplayListener);
 
-    const promise = codeReader.decodeFromVideoDevice(
-      undefined,
-      videoElement,
-      (result, error) => {
-        if (result) {
-          handleScan(result.getText());
-        }
-        if (error) {
-          handleError(error.message);
-        }
-      }
-    );
+    const pendingScanRequest =
+      videoElement?.pendingScanRequest ?? Promise.resolve(undefined);
+    const scanRequest = pendingScanRequest
+      .then(() =>
+        codeReader.decodeFromVideoDevice(
+          undefined,
+          videoElement,
+          (result, error) => {
+            if (result) {
+              handleScan(result.getText());
+            }
+            if (error) {
+              handleError(error.message);
+            }
+          }
+        )
+      )
+      .catch((error) => {
+        console.error(error);
+        return undefined;
+      });
```

Comment:
Moved the catch here for better consistency. This way `scanRequest` doesn't throw and only returns a `IScannerControls | undefined` value.

Also, maybe the error should be escalated to the component's consumer (not sure if the `handleError` callback would work)